src/test/tv/codely/scala_http_api/module/video/application/search/VideosSearcherTest.scala

```diff
@@ -0,0 +1,20 @@
+package tv.codely.scala_http_api.module.video.application.search
+
+import tv.codely.scala_http_api.module.video.VideoUnitTestCase
+import tv.codely.scala_http_api.module.video.domain.VideoStub
+
+final class VideosSearcherTest extends VideoUnitTestCase {
+  private val searcher = new VideosSearcher(repository)
+
+  "Videos Searcher" should {
+    "search all existing videos" in {
+      val existingVideo        = VideoStub.random
+      val anotherExistingVideo = VideoStub.random
+      val existingVideos       = Seq(existingVideo, anotherExistingVideo)
+
+      shouldSearchAllVideos(existingVideos)
```

Comment:
Same as with the `shouldSaveVideo` comment

Reply:
`shouldSearchAllVideosInVideoRepository`?

Reply:
Given we only have 1 repo/module -> `shouldSearchAllVideosInRepository`

Reply:
And given we only have 1 AG/module -> `shouldSearchAllInRepository`